Sources/TuistCore/NodeLoaders/FrameworkNodeLoader.swift

```diff
@@ -49,10 +62,28 @@ public final class FrameworkNodeLoader: FrameworkNodeLoading {
         let linking = try frameworkMetadataProvider.linking(binaryPath: binaryPath)
         let architectures = try frameworkMetadataProvider.architectures(binaryPath: binaryPath)
 
+        let dependencies = try dlybDependenciesPath(forBinaryAt: binaryPath)
+
         return FrameworkNode(path: path,
                              dsymPath: dsymsPath,
                              bcsymbolmapPaths: bcsymbolmapPaths,
                              linking: linking,
-                             architectures: architectures)
+                             architectures: architectures,
+                             dependencies: dependencies)
+    }
+
+    private func dlybDependenciesPath(forBinaryAt binaryPath: AbsolutePath) throws -> [PrecompiledNode.Dependency] {
+        try otoolController
+            .dlybDependenciesPath(forBinaryAt: binaryPath)
+            .toBlocking()
+            .first()?
+            .filter { $0 != binaryPath }
+            .map { $0.removingLastComponent() }
+            .map { dependencyPath -> PrecompiledNode.Dependency in
```

Comment:
I see 2 issues here:
- With `removingLastComponent` and `load(path: dependencyPath)` we are assuming that the dependency is a framework, and that might not be true. I'd recommend checking whether the dependency is a framework or a library, and use either `FrameworkNodeLoader` or `LibraryNodeLoader`.
- Since we are recursively building the graph, circular dependencies might lead to a loop in Tuist. For that reason, I recommend moving this logic to [here](https://github.com/tuist/tuist/blob/149a442dbfa58d6ad1218a18b817b59be61cae8a/Sources/TuistCore/Graph/GraphLoader.swift#L254), because that loader is aware of the loading process and can re-use any previously loaded framework.